gpytorch/priors/prior.py

```diff
@@ -15,10 +15,9 @@ class Prior(Distribution, Module, ABC):
     respective prior in computing the Marginal Log-Likelihood.
     """
 
-    @property
-    def log_transform(self):
-        return self._log_transform
+    def transform(self, parameter):
+        return self._transform(parameter) if self._transform is not None else parameter
```

Comment:
Priors no longer need a `transform` function, right? Since the transform is being handled when you register a prior.

Reply:
So the thinking is that this transform can be independent of the parameter transform. The idea is that using this you can easily instantiate, say, a log-normal prior, independent of registering it anywhere. This should help separate concerns and compartmentalize the design. If you think this is an unnecessary complication this could be removed, then in order to do this the transform would have to be included in the transform registered with this prior (and you couldn't have a standalone log-normal `Prior` object without implementing a new class)